### PR TITLE
Added commandlet to try finding path to preinstalled dotnet.exe

### PR DESCRIPTION
**Pull request by agr**
Need `dotnet` to run tests in some repos, added discovery.

### CODE REVIEW

File: build/common.ps1 [hunk 1]
```diff
@@ -543,6 +543,22 @@ Function Install-DotnetCLI {
     & $DotNetExe --info
 }
 
+Function Get-ExistingDotnet {
+    $p = Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\ASP.NET Core\Shared Framework"
```
Comment (joelverhagen) [inline build/common.ps1:547]: Is this the recommended approach from .NET CLI? The "ASP.NET Core" part looks a little suspicious/specific given the context

**Review by zhhyu — APPROVED** [review]

File: build/common.ps1 (hunk 1 above)
Comment (agr) [reply build/common.ps1:547]: I didn't find an "official prescription" for discovery. That's what I found to work on my and build machines we use. I will ask dotnet folks about it.

Comment (joelverhagen) [reply build/common.ps1:547]: Okay, if it works for us, don't sink too much time of the "proper". This can be easily fixed later if need be.

**Review by joelverhagen — APPROVED** [review]

**Comment by agr** [comment]
Will imply that it is available through the `PATH` environment variable for now.

**Pull request closed without merging**